### ISSUE TITLE
Item.publish does not return Future immediately

### ISSUE THREAD

**Issue by rkraujutis** [Esri/arcgis-python-api#2152]
**Describe the bug**

`Item.publish` method with `future = True` parameter does not return future immediately, but waits future to finish.

**To Reproduce**

```python
        sd_item:Item = self.content.get("391ca075826f4fb9844eaa93f61b25f9")
        start_time = datetime.now()
        sd_item.publish(
            overwrite = True,
            future = True, # should not wait for the job to complete, but return future immediately
        )
        duration = (datetime.now() - start_time).total_seconds()
        assert duration < 10, f"Publishing took too long: {duration} seconds"
```
error: script hangs and waits publish job to be finished.

**Expected behavior**

Expected that Future object is returned and waiting could be done by me.

**Platform (please complete the following information):**
- OS: Linux
- Browser: -
- Python API Version: 2.4.0

**Additional context**

Seems, that issue is in `arcgis/gis/__init__.py` file 16270 line, where code submits task and in line 16271 waits for thread pool to finish:
```python
        job: concurrent.futures.Future = tp.submit(self._publish, **params)
        tp.shutdown(wait=True)
```

**Comment by nanaeaubry** [comment]
@rkraujutis Thanks for reporting, we will take a look

**Comment by nanaeaubry** [comment]
This has been fixed for the next release: v2.4.1 scheduled for beginning 2025.
We don't have a workaround to suggest as of now, thank you again for reporting
